### PR TITLE
[.github] - refacto: specify deploy region 

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at passing the region of the Artifact Registry repository in which to push the image. It also modifies the `deploy-prodbox` workflow to support multi-region deployment.

Note: the multi-region deployment is experimental and needs to be merged on main to be tested. Once fully working for prodbox I will migrate the other deployments to follow the same logic.

## Risk

Breaking deployment workflow

### CODE REVIEW

File: .github/workflows/deploy-alerting-temporal.yml [hunk 1]
```diff
@@ -44,7 +44,8 @@ jobs:
             --image-name=alerting-temporal \
             --dockerfile-path=./Dockerfile \
             --working-dir=./alerting/temporal/ \
-            --dust-client-facing-url=https://dust.tt
+            --dust-client-facing-url=https://dust.tt \ 
+            --region=us-central1
```
Comment (JulesBelveze) [inline .github/workflows/deploy-alerting-temporal.yml:48]: Hardcoded for now until matrix version is fully working

File: k8s/cloud-build.sh [hunk 2]
```diff
@@ -54,13 +59,13 @@ cd "$WORKING_DIR"
...
 # prepare substitutions
-SUBSTITUTIONS="SHORT_SHA=$(git rev-parse --short HEAD),_IMAGE_NAME=$IMAGE_NAME,_DOCKERFILE_PATH=$DOCKERFILE_PATH"
+SUBSTITUTIONS="SHORT_SHA=$(git rev-parse --short HEAD),_IMAGE_NAME=$IMAGE_NAME,_DOCKERFILE_PATH=$DOCKERFILE_PATH,_REGION=$REGION"
 if [ -n "$DUST_CLIENT_FACING_URL" ]; then
     SUBSTITUTIONS="$SUBSTITUTIONS,_DUST_CLIENT_FACING_URL=$DUST_CLIENT_FACING_URL"
 fi
 
 # prepare the gcloud command
-GCLOUD_CMD=(gcloud builds submit --quiet --config "${SCRIPT_DIR}/cloudbuild.yaml")
+GCLOUD_CMD=(gcloud builds submit --quiet --config "${SCRIPT_DIR}/cloudbuild.yaml" --region="$REGION")
```
Comment (JulesBelveze) [inline k8s/cloud-build.sh:68]: The --region flag specifies in which region to build the image (not linked to the artifact registry region but using the same might speed things up)

**Comment by JulesBelveze** [comment]
PMRR

File: .github/workflows/deploy-front-edge.yml [hunk 3]
```diff
@@ -3,6 +3,15 @@ name: Deploy Front Edge
 on:
   workflow_dispatch:
     inputs:
+      regions:
+        description: "Regions to deploy to"
```
Comment (spolu) [inline .github/workflows/deploy-front-edge.yml:7]: Eventually we'll want to deploy to both regions right?

Comment (JulesBelveze) [reply .github/workflows/deploy-front-edge.yml:7]: Yup, added a "all" option in a next PR 😃

**Pull request merged**